aesd-char-driver/main.c

```diff
@@ -146,12 +154,134 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
     mutex_unlock(&dev->lock);
     return retval;
 }
+
+loff_t aesd_llseek(struct file *filp, loff_t offset, int whence)
+{
+    struct aesd_dev *dev = NULL;
+    loff_t file_offset = 0;
+    uint8_t index = 0;
+    struct aesd_buffer_entry *entry = NULL;
+    loff_t total_size = 0;
+
+    if (NULL == filp)
+    {
+        PDEBUG("ERROR: aesd_llseek invalid arguments");
+        return -EINVAL;
+    }
+
+    dev = filp->private_data;
+
+    if (0 != mutex_lock_interruptible(&dev->lock))
+    {
+        PDEBUG("ERROR: mutex_lock_interruptible acquiring lock");
+        return -ERESTARTSYS;
+    }
+    AESD_CIRCULAR_BUFFER_FOREACH(entry,&aesd_device.buffer,index)
+    {
+        total_size += entry->size;
+    }
+    mutex_unlock(&dev->lock);
+
+    file_offset = fixed_size_llseek(filp, offset, whence, total_size);
+    
+    return file_offset;
+
+}
+
+static long aesd_adjust_file_offset(struct file *filp, unsigned int write_cmd,
+                                    unsigned int write_cmd_offset)
+{
+    struct aesd_dev *dev = NULL;
+    long return_value = 0;
+    uint8_t index = 0;
+    struct aesd_buffer_entry *entry = NULL;
+
+    if (NULL == filp)
+    {
+        PDEBUG("ERROR: aesd_adjust_file_offset invalid arguments");
+        return -EINVAL;
+    }
+
+    dev = filp->private_data;
+
+    if (0 != mutex_lock_interruptible(&dev->lock))
+    {
+        PDEBUG("ERROR: mutex_lock_interruptible acquiring lock");
+        return -ERESTARTSYS;
+    }
+    /* Get the total entries count in circular buffer in index */
+    AESD_CIRCULAR_BUFFER_FOREACH(entry,&aesd_device.buffer,index)
```

Comment:
I think this block could be removed as well as the `(write_cmd > index)` check below, since index will always be `AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED`

Reply:
Sure, thanks for the review.